service/src/main/kotlin/io/provenance/digitalcurrency/consortium/service/TxBatchHandler.kt

```diff
@@ -0,0 +1,99 @@
+
+@Service
+class TxBatchHandler(
+    private val pbcService: PbcService,
+    private val provenanceProperties: ProvenanceProperties,
+    private val bankClientProperties: BankClientProperties,
+    private val mapper: ObjectMapper,
+    private val pbcTimeoutService: PbcTimeoutService,
+) {
+    private val log = logger()
+
+    @Scheduled(
+        initialDelayString = "\${queue.batch_initial_delay_ms}",
+        fixedRateString = "\${queue.batch_polling_delay_ms}"
+    )
+    fun batchQueuedTxns() {
+        transaction {
+            TxRequestViewRecord.findQueued().mapNotNull { request ->
+                when (request.type) {
+                    TxRequestType.MINT -> CoinMintRecord.findById(request.id)!!.let {
+                        it to buildExecuteContractMessage(it.getExecuteContractMessage())
+                    }
+                    TxRequestType.REDEEM -> CoinRedemptionRecord.findById(request.id)!!.let {
+                        it to buildExecuteContractMessage(it.getExecuteContractMessage(bankClientProperties.denom))
+                    }
+                    TxRequestType.BURN -> CoinBurnRecord.findById(request.id)!!.let {
+                        it to buildExecuteContractMessage(it.getExecuteContractMessage())
+                    }
+                    TxRequestType.TAG -> AddressRegistrationRecord.findById(request.id)!!.let {
+                        if (!tagExists(it.address)) {
+                            it to it.getAddAttributeMessage(pbcService.managerAddress, bankClientProperties.kycTagName)
+                        } else {
+                            // technically should not happen
+                            log.warn("Address ${it.id} already tagged - completing")
+                            it.status = TxStatus.COMPLETE
+                            null
+                        }
+                    }
+                    TxRequestType.DETAG -> AddressDeregistrationRecord.findById(request.id)!!.let {
+                        if (tagExists(it.addressRegistration.address)) {
+                            it to it.getDeleteAttributeMessage(
+                                pbcService.managerAddress,
+                                bankClientProperties.kycTagName
+                            )
+                        } else {
+                            // technically should not happen
+                            log.warn("Address ${it.id} already de-tagged - completing")
+                            it.status = TxStatus.COMPLETE
+                            null
+                        }
+                    }
+                }
+            }.let {
+                val (baseRequestRecords, messages) = it.unzip()
+                val timeoutHeight = pbcTimeoutService.getBlockTimeoutHeight()
+                try {
+                    val response = pbcService.broadcastBatch(messages, timeoutHeight)
+                    baseRequestRecords.forEach { baseRequestRecord ->
```

Comment:
if this can be separate tx, then we can do a repeat around it since the worst case scenario is batch succeeds submission but we arent able to update the db